playbooks/gce/openshift-cluster/config.yml

```diff
@@ -15,10 +15,10 @@
 
 - include: ../../common/openshift-cluster/config.yml
   vars:
-    g_etcd_group: "{{ 'tag_env-host-type-' ~ cluster_id ~ '-openshift-etcd' }}"
-    g_lb_group: "{{ 'tag_env-host-type-' ~ cluster_id ~ '-openshift-lb' }}"
-    g_masters_group: "{{ 'tag_env-host-type-' ~ cluster_id ~ '-openshift-master' }}"
-    g_nodes_group: "{{ 'tag_env-host-type-' ~ cluster_id ~ '-openshift-node' }}"
+    g_etcd_group: "{{ groups['tag_host-type-openshift-etcd'] | intersect(groups['tag_environment_' ~ cluster_id}}"
+    g_lb_group: "{{ 'tag_host-type-openshift-lb' | intersect(groups['tag_environment_' ~ cluster_id] }}"
+    g_masters_group: "{{ 'tag_host-type-openshift-master' | intersect(groups['tag_environment_' ~ cluster_id] }}"
+    g_nodes_group: "{{ 'tag_host-type-openshift-node' | intersect(groups['tag_environment_' ~ cluster_id] }}"
```

Comment:
If we are going to change g_*_group to be a list of hosts instead of a group name, then I'd rather that we rename them to g_*_hosts.

As an aside, we'll need to update the other playbooks (byo, openstack, libvirt) as well for these changes, otherwise they will be broken with this change.

Another thing to consider is that with the gce support of tag_<tag-name>_<tag_val> group names, we could further consolidate the aws, gce, openstack and libvirt config playbooks to share a common cloud provider set of playbooks that can do things like set tags to be used, group names based on tags and other common variables. The libvirt provider playbooks would need to be updated to change the names to use the same tag_<tag-name>_<tag_val> format to reach that though.